recommend/policyTemplates.go

```diff
@@ -0,0 +1,247 @@
+	latestRelease, err := latestRelease()
+	if err != nil {
+		return "", err
+	}
+	_ = removeData(path)
+	err = os.MkdirAll(filepath.Dir(path), 0750)
+	if err != nil {
+		return "", err
+	}
+	downloadURL := fmt.Sprintf("%s%s.zip", url, *latestRelease.TagName)
+	resp, err := grab.Get(path, downloadURL)
+	if err != nil {
+		_ = removeData(path)
+		return "", err
+	}
+	err = unZip(resp.Filename, path)
+	if err != nil {
+		return "", err
+	}
+	err = removeData(resp.Filename)
+	if err != nil {
+		return "", err
+	}
+	_ = updatePolicyRules(strings.TrimSuffix(resp.Filename, ".zip"))
+	return *latestRelease.TagName, nil
+}
+
+func unZip(source, dest string) error {
+	read, err := zip.OpenReader(source)
+	if err != nil {
+		return err
+	}
+	defer read.Close()
+	for _, file := range read.File {
+		if file.Mode().IsDir() {
+			continue
+		}
+		open, err := file.Open()
+		if err != nil {
+			return err
+		}
+		name, err := sanitizeArchivePath(dest, file.Name)
+		if err != nil {
+			return err
+		}
+		_ = os.MkdirAll(path.Dir(name), 0750)
+		create, err := os.Create(filepath.Clean(name))
+		if err != nil {
+			return err
+		}
+		_, err = create.ReadFrom(open)
+		if err != nil {
+			return err
+		}
+		if err = create.Close(); err != nil {
+			return err
+		}
+		defer open.Close()
+	}
+	return nil
+}
+
+func updatePolicyRules(filePath string) error {
+	var files []string
+	err := filepath.Walk(filePath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() && info.Name() == "metadata.yaml" {
+			files = append(files, path)
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	f, err := os.Create(filepath.Clean(fmt.Sprintf("%s/.cache/karmor/rules.yaml", userHome())))
+	if err != nil {
+		log.WithError(err).Error(fmt.Sprintf("create file %s failed", fmt.Sprintf("%s/.cache/karmor/rules.yaml", userHome())))
+	}
+
+	idx := 0
+	var yamlFile []byte
+	for _, file := range files {
+		yamlFile, err = os.ReadFile(filepath.Clean(file))
+		if err != nil {
+			return err
+		}
+		updateRulesYAML(yamlFile)
+		ms, err := getNextRule(&idx)
+		for ; err == nil; ms, err = getNextRule(&idx) {
+			tempYaml := yamlFile
+			if ms.Yaml != "" {
+
+				newYaml, err := os.ReadFile(filepath.Clean(fmt.Sprintf("%s%s", strings.TrimSuffix(file, "metadata.yaml"), ms.Yaml)))
+				if err != nil {
+					return err
+				}
+				skipCount := len(ms.Yaml) + 6
+				tempYaml = tempYaml[strings.Index(string(tempYaml), "yaml:")+skipCount:]
+				dataVal := strings.TrimSpace(string(tempYaml))
+				yamlFile = yamlFile[:strings.Index(string(yamlFile), "yaml:")-1]
+				for _, eachLine := range strings.Split(string(newYaml[strings.Index(string(newYaml), "spec:"):]), "\n") {
+					yamlFile = append(yamlFile, []byte(fmt.Sprintf(" %s\n", eachLine))...)
+				}
+				yamlFile = []byte(strings.TrimSpace(string(yamlFile)))
+				yamlFile = append(yamlFile, []byte(fmt.Sprintf("\n%s", dataVal))...)
+			}
```

Comment:
We shouldn't get our hands dirty by parsing the YAML file manually. Each `ms.Yaml` should be a valid KubeArmorPolicy file, right? So, use `KubeArmorPolicy` struct from [the package](https://pkg.go.dev/github.com/kubearmor/KubeArmor/pkg/KubeArmorPolicy/api/security.kubearmor.com/v1), do `yaml.Unmarshall` and extract the `spec` from the policy.

Reply:
Since all the policies in policy-template start with a comment, it's showing JSON unmarshal error `error="invalid character '#' looking for beginning of value"` @wazir-ahmed